src/bp/ui-studio/src/web/components/Layout/BottomPanel/Debugger/components/Settings.tsx

```diff
@@ -0,0 +1,74 @@
+import { Button, FormGroup, InputGroup, Intent, TextArea } from '@blueprintjs/core'
+import { lang, toast } from 'botpress/shared'
+import React, { useEffect, useState } from 'react'
+
+import style from './style.scss'
+import ConfigEditor from './ConfigEditor'
+import RawPayloadSender from './RawPayloadSender'
+
+const Settings = props => {
+  const [userId, setUserId] = useState('')
+  const [externalAuthToken, setExternalAuthToken] = useState('')
+
+  useEffect(() => {
+    if (props.config) {
+      const parsed = JSON.parse(props.config)
+
+      setUserId(parsed.userId)
+      setExternalAuthToken(parsed.externalAuthToken)
+    }
+  }, [props.config])
+
+  if (!props.config) {
+    return null
+  }
+
+  const applyBasicConfig = () => {
+    try {
+      window.botpressWebChat.mergeConfig({ userId, externalAuthToken })
```

Comment:
What's the difference between this and saveConfig in ConfigEditor ?

Reply:
The userId changed here is permanent while the other is temporary. I wasn't sure about keeping both (basic and advanced) config, so i'm open for discussion about it